GrayBMP.cs

```diff
@@ -126,6 +125,39 @@ public void Dirty ()
       End ();
    }
 
+   /// <summary>
+   /// Draws a thick line between the two given end-points (with given size & given shade of gray)
+   /// </summary>
+
+   public void DrawThickLine (int x1, int y1, int x2, int y2, int width, int color) {
+      Begin (); 
+      Check (x1, y1); Check (x2, y2); Dirty (x1, y1, x2, y2);
+
+      const double OneRadian = PI / 180;
+      double offset = width * 0.5;
+      var BaseAngle = Atan2 (y2 - y1, x2 - x1) / OneRadian;
+
+      System.Drawing.Point[] HexPoints = new System.Drawing.Point[8];
+      for (int Theta1 = 90, Theta2 = 270, i = 0; Theta1 <= 270; Theta1 += 60, Theta2 += 60, i++) {
+         var A = Translate (x1, y1, Theta1 + BaseAngle, offset);
+         var B = Translate (x2, y2, Theta2 + BaseAngle, offset);
+         HexPoints[i] = A; HexPoints[i + 4] = B;
+      }

```

Comment:
Instead of first creating a point and then looping again to draw line. Try to do it in single pass.